App.js

```diff
@@ -300,6 +314,63 @@ export default class App extends Component {
     }
   }
 
+  getOnlineInfo = () => {
+    //let this = this;
+    this.pubnub.hereNow({
+      includeUUIDs: true,
+    },
+     (status, response) => {
+      let uuids = [];
+      console.log(response)
+      let online = response.channels.global.occupants;
+      for( i in online){
+        uuids.push(online[i].uuid)
+      }
+      let users = this.state.users;
+      let loopCount = null;
+      while(uuids.length != 0 && loopCount < 10){
+        let timetoken = "0";
+        this.pubnub.history({
+          channel: 'global',
+          start: null,
+          stringifiedTimeToken: true // false is the default
+          
+        }, (status, response) => {
+          timetoken = response.startTimeToken;
+          
+          for(i in response.messages){
+            let index = uuids.indexOf(response.messages[i].entry.uuid);
```

Comment:
please time how long it takes to iterate through 1000 messages; or make this run asynchronously